accepted/00076-configuration-channels-visibility.md

```diff
@@ -0,0 +1,201 @@
+After:
+![System Groups AFTER](images/configuration-channels-visibility/Systems_SystemGroups_Details_States_ConfigurationChannels_AFTER.png)
+
+### `Systems > System Set Manager > Overview > Configuration > Subscribe to Channels`
+Before:
+![SSM Subscribe To Channels BEFORE](images/configuration-channels-visibility/Systems_SSM_Configuration_SubscribeToChannels_BEFORE.png)
+
+After:
+![SSM Subscribe To Channels AFTER](images/configuration-channels-visibility/Systems_SSM_Configuration_SubscribeToChannels_AFTER.png)
+
+### `Systems > System Set Manager > Overview > Configuration > Unsubscribe to Channels`
+Before:
+![SSM Unsubscribe From Channels BEFORE](images/configuration-channels-visibility/Systems_SSM_Configuration_UnsbscribeFromChannels_BEFORE.png)
+
+After:
+![SSM Unsubscribe From Channels AFTER](images/configuration-channels-visibility/Systems_SSM_Configuration_UnsubscribeFromChannels_AFTER.png)
+
+### `Configuration > Overview`
+Before:
+![Configuration Overview BEFORE](images/configuration-channels-visibility/Configuration_Overview_BEFORE.png)
+
+After:
+![Configuration Overview AFTER](images/configuration-channels-visibility/Configuration_Overview_AFTER.png)
+
+### `Configuration > Channels`
+Before:
+![Configuration Channels BEFORE](images/configuration-channels-visibility/Configuration_Channels_BEFORE.png)
+
+After:
+![Configuration Channels AFTER](images/configuration-channels-visibility/Configuration_Channels_AFTER.png)
+
+### `Configuration > Channels > Details > Overview`
+Before:
+![Configuration Channels Detail Overview BEFORE](images/configuration-channels-visibility/Configuration_Channels_Details_Overview_BEFORE.png)
+
+After:
+![Configuration Channels Detail Overview AFTER](images/configuration-channels-visibility/Configuration_Channels_Details_Overview_AFTER.png)
+
+### `Configuration > Channels > Details > Systems > Inheriting Systems`
+This is a new page where systems are listed only if not subscribed to the channel directly, but inheriting it through Organization or System Group assignment.
+
+Before:
+![Configuration Channels Detail Systems BEFORE](images/configuration-channels-visibility/Configuration_Channels_Details_Systems_BEFORE.png)
+
+After:
+![Configuration Channels Detail Systems AFTER](images/configuration-channels-visibility/Configuration_Channels_Details_Systems_AFTER.png)
+
+
+### `Configuration > Channels > Details > System Groups`
+This page reproduces the same behavior of the `Configuration > Channels > Details > Systems` tab, but for System Groups.
+
+This **new** page is pointed by:
+- the `Channel information` box link of the system groups counting in the `Configuration > Channels > Details > Overview` page: the link will land on the new page at the `Subscribed System Groups` tab
+
+### `Configuration > Channels > System Groups`
+This page represents a complete list of pairs value `Configuration State Channel - System Group`: every value is a link to the specific entity detail (channel or system group).
```

Comment:
I'm not sure if the added value of such a über-overview page is bigger than the possible confusion in brings. If the user has a lot of groups assigned to a lot of channels, then the list grows pretty well: if you have 5 groups and each one has 10 channels assigned, then the list will have 50 items. Then the user would need to have some kind of filtering in order to focus on needed groups/channels. But then again: they could use the channel details or system group details pages for getting the same information.
But maybe there is a use case that I overlooked.

Reply:
Thanks for raising this up: your last point is exactly what this page would be there to serve. At the moment in case you want look for an entity, you need to go **either** to the channel details page **or** the system group details one. While it is true that the most common (probably the only one) use case is to look for one entry at a time, at the same time the UX would be a PITA if you want to look first for more than one channel (you would have to go back and forth from channel details to the right tab and back to the list and forth again), or maybe you first want to look for a channel because you want to know all the groups connected, then for a system group because you want also to check about all the channels connected to it, just to make sure you have everything under your radar. In order to achieve that, you would have to go crazy around pages, while with a single page you could have all, just a matter of filtering. For sure filters are meant to be there for both the channels and the system groups.

Overall, this is one of a generic issue that exists in the product for the most of the features: doing slightly the same (but not exactly the same) in many different places, but no place to do them all.

There is one possible drawback, and I'd like to hear your POV @hustodemon : could it be a slow performing page presenting a huge amount of data? I don't really see this as an issue because it's just a list of pair values, but maybe I am missing something in the picture.

Reply:
Ok, I agree, that having a single page with a responsive filtering, where the user could just punch in the channel name or the system group name would be a nice feature.

I suppose this will be implemented as a eager fetch from the backend and then client-side filtering (afaik, we still don't have a server-side filtering for the react tables). In this case, the data set would need to be really big to exhibit low performance. The main problem I have is more general: it's the absence of the server-side filtering "framework" for the new components. What we're doing is that we are piling new components that work ok on reasonable data sets to Uyuni. Then one day, when we discover we need to rewrite them all, it can be to much. So in the end, this is a tech debt problem and we should talk about it outside this RFC.